File: src/main/java/com/hemanth/mockito/annotations/support/BookService.java
package com.hemanth.mockito.annotations.support;

import java.util.List;

public class BookService {

    private BookRespository bookRespository;

    public BookService(BookRespository bookRespository) {
        this.bookRespository = bookRespository;
    }

    public List<Book> getNewBooksWithAppliedDiscount(int discountRate, int days){
        List<Book> newBooks = bookRespository.findNewBooks(days);

        for (Book book : newBooks){
            int price = book.getPrice();
            int newPrice = price - ((discountRate * price) / 100);
            book.setPrice(newPrice);
        }

        return newBooks;
    }
}

File: src/main/java/com/hemanth/mockito/stubbing/BookService.kt
package com.hemanth.mockito.stubbing

import java.time.LocalDate

class BookService(
    val bookRepository: BookRepository
) {

    fun getNewBooksWithAppliedDiscount(discountRate: Int, days: Int): List<Book> {
        var newBooks: List<Book> = bookRepository.findNewBooks(days)

        for (book in newBooks) {
            val price = book.price
            val newPrice = price - discountRate * price / 100
            book.price = newPrice
        }
        return newBooks
    }

    fun calculateTotalConst(bookIds: List<String>): Int {
        var total: Int = 0

        for (bookId in bookIds) {
            var book: Book = bookRepository.findBookByBookID(bookId)
            total = total + book.price
        }

        return total
    }

    fun addBook(book: Book) {
        bookRepository.save(book)
    }

    fun addBook(bookRequest: BookRequest) {
        var book = Book(null, bookRequest.title, bookRequest.price, LocalDate.now())
        bookRepository.save(book)

    }

}